### DISCUSSION TITLE
Does the spec indictate how collections and other values should be rendered?

### DISCUSSION THREAD

**Discussion by dpeterson-git** [mustache/spec#183] [Q&A] [answered]
Consider a context object like this (in JSON notation):

```
{"A": [1, ["2", 3.0], {"D": 4}, 5.0, null, true, false, -314e-2]}
```

and a Mustache template like:

```
{{A}}
```

Most JavaScript-based implementations produce output like:
```
1,2,3,[object Object],5,,true,false,-3.14
```
However [a prominent Java-based implementation](https://github.com/spullara/mustache.java) produces output like:
```
[1, [2, 3.0], {D&#61;4}, 5.0, null, true, false, -3.14]
```

Notable differences include:

1. Arrays are rendered with and without leading `[` and trailing `]` characters.
2. Items in the list/array are separated with `, ` vs `,`.
3. Objects are displayed as `[object Object]` vs `{D&#61;4}`.
4. The null value displays as the empty string vs `null`.
5. Floating-point numbers are displayed as `5` vs `5.0`.

Does the spec offer guidance on how these items _should_ be rendered, or is this left up to implementations?

**Comment by jgonggrijp** [comment, marked as answer]
How to render collections is left implementation-defined:

https://github.com/mustache/spec/blob/83b221aec50adb91eeaa935ba7bd51cd792daece/specs/interpolation.yml#L23-L24

That being said, null should interpolate as the empty string:

https://github.com/mustache/spec/blob/83b221aec50adb91eeaa935ba7bd51cd792daece/specs/interpolation.yml#L104-L120

**Reply by dpeterson-git** [reply]
Understood.  Thank you for the clarification.
